src/devices/HardwareMonitor/OpenHardwareMonitorWmi.cs

```diff
@@ -0,0 +1,250 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Management;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using HardwareMonitor;
+using UnitsNet;
+using UnitsNet.Units;
+
+// We have a check on all constructors to ensure that we throw when not on Windows, so disabling warning on calling windows-only apis.
+#pragma warning disable CA1416 // Validate platform compatibility
+
+namespace Iot.Device.HardwareMonitor
+{
+    /// <summary>
+    /// This class connects to a running instance of OpenHardwareMonitor and reads out all available values using WMI.
+    /// This works only if OpenHardwareMonitor (https://openhardwaremonitor.org/) is currently running.
+    /// While the tool needs to be run with elevated permissions, the application using this binding does not.
+    /// The connection using WMI has proven to be unreliable (the WMI classes are sometimes just not visible), therefore
+    /// OHM versions greater > 0.10.0 use a rest api instead.
+    /// </summary>
+    internal sealed class OpenHardwareMonitorWmi : IDisposable, IOpenHardwareMonitorInternal
```

Comment:
considering this is legacy what's the benefit of supporting this?

Reply:
It would work with the officially released version of OHM. The new transport requires a new build, which should be available shortly.